dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1343,6 +1343,49 @@ void TeachingTip::ClosePopup()
     }
 }
 
+winrt::TeachingTipPlacementMode TeachingTip::GetFlowDirectionAdjustedPlacement(const winrt::TeachingTipPlacementMode& placementMode)
+{
+    if (FlowDirection() == winrt::FlowDirection::LeftToRight)
+    {
+        return placementMode;
+    }
+    else
+    {
+        switch (placementMode)
+        {
+            case winrt::TeachingTipPlacementMode::Auto:
+                return winrt::TeachingTipPlacementMode::Auto;
+            case winrt::TeachingTipPlacementMode::Left:
+                return winrt::TeachingTipPlacementMode::Right;
+            case winrt::TeachingTipPlacementMode::Right:
+                return winrt::TeachingTipPlacementMode::Left;
+            case winrt::TeachingTipPlacementMode::Top:
+                return winrt::TeachingTipPlacementMode::Top;
+            case winrt::TeachingTipPlacementMode::Bottom:
+                return winrt::TeachingTipPlacementMode::Bottom;
+            case winrt::TeachingTipPlacementMode::LeftBottom:
+                return winrt::TeachingTipPlacementMode::RightBottom;
+            case winrt::TeachingTipPlacementMode::LeftTop:
+                return winrt::TeachingTipPlacementMode::RightTop;
+            case winrt::TeachingTipPlacementMode::TopLeft:
+                return winrt::TeachingTipPlacementMode::TopRight;
+            case winrt::TeachingTipPlacementMode::TopRight:
+                return winrt::TeachingTipPlacementMode::TopLeft;
+            case winrt::TeachingTipPlacementMode::RightTop:
+                return winrt::TeachingTipPlacementMode::LeftTop;
+            case winrt::TeachingTipPlacementMode::RightBottom:
+                return winrt::TeachingTipPlacementMode::LeftBottom;
+            case winrt::TeachingTipPlacementMode::BottomRight:
+                return winrt::TeachingTipPlacementMode::BottomLeft;
+            case winrt::TeachingTipPlacementMode::BottomLeft:
+                return winrt::TeachingTipPlacementMode::BottomRight;
+            case winrt::TeachingTipPlacementMode::Center:
+                return winrt::TeachingTipPlacementMode::Center;
+        }
+    }
+    return winrt::TeachingTipPlacementMode::Auto;
```

Comment:
>TeachingTipPlacementMode [](start = 18, length = 24)

Is this needed?

Reply:
Either this or a default cause for our switch case, otherwise not all code paths would return a value according to the compiler.